sse2neon.h

```diff
@@ -8358,6 +8359,658 @@ FORCE_INLINE int _mm_testz_si128(__m128i a, __m128i b)
 
 /* SSE4.2 */
 
+typedef enum {
+    _MM_CMPINT_EQ = 0,    /* Equal */
+    _MM_CMPINT_LT = 1,    /* Less than */
+    _MM_CMPINT_LE = 2,    /* Less than or Equal */
+    _MM_CMPINT_FALSE = 3, /* Always False */
+    _MM_CMPINT_NE = 4,    /* Not Equal */
+    _MM_CMPINT_NLT = 5,   /* Not Less than */
+    _MM_CMPINT_NLE = 6,   /* Not Less than or Equal */
+    _MM_CMPINT_TRUE = 7   /* Always True */
+} _MM_CMPINT_ENUM;
+
+static uint16_t mask_for_16bit_elements[8] __attribute__((aligned(16))) = {
+    0x01, 0x02, 0x04, 0x08, 0x10, 0x20, 0x40, 0x80,
+};
+static uint8_t mask_for_8bit_elements[16] __attribute__((aligned(16))) = {
+    0x01, 0x02, 0x04, 0x08, 0x10, 0x20, 0x40, 0x80,
+    0x01, 0x02, 0x04, 0x08, 0x10, 0x20, 0x40, 0x80,
+};
+
+/* specify the source data format */
+#define _SIDD_UBYTE_OPS 0x00 /* unsigned 8-bit characters */
+#define _SIDD_UWORD_OPS 0x01 /* unsigned 16-bit characters */
+#define _SIDD_SBYTE_OPS 0x02 /* signed 8-bit characters */
+#define _SIDD_SWORD_OPS 0x03 /* signed 16-bit characters */
+
+/* specify the comparison operation */
+#define _SIDD_CMP_EQUAL_ANY 0x00     /* compare equal any: strchr */
+#define _SIDD_CMP_RANGES 0x04        /* compare ranges */
+#define _SIDD_CMP_EQUAL_EACH 0x08    /* compare equal each: strcmp */
+#define _SIDD_CMP_EQUAL_ORDERED 0x0C /* compare equal ordered */
+
+/* specify the polarity */
+#define _SIDD_POSITIVE_POLARITY 0x00
+#define _SIDD_MASKED_POSITIVE_POLARITY 0x20
+#define _SIDD_NEGATIVE_POLARITY 0x10 /* negate results */
+#define _SIDD_MASKED_NEGATIVE_POLARITY \
+    0x30 /* negate results only before end of string */
+
+/* specify the output selection in _mm_cmpXstri */
+#define _SIDD_LEAST_SIGNIFICANT 0x00
+#define _SIDD_MOST_SIGNIFICANT 0x40
+
+/* specify the output selection in _mm_cmpXstrm */
+#define _SIDD_BIT_MASK 0x00
+#define _SIDD_UNIT_MASK 0x40
+
+#define PCMPSTR_EQ_16x8(a, b, mtx)                                      \
```

Comment:
Alternatively, we can make use of [X macro](https://en.wikipedia.org/wiki/X_Macro) technique to simplify the manual loop unrolling and corresponding logic.